### DISCUSSION TITLE
Constraint force differentiability?

### DISCUSSION THREAD

**Discussion by gpu-daal** [google/brax#602] [General]
It seems like the entire rigid-body solver is differentiable, but I was wondering how constraint force calculation is made differentiable. AFAIK, it's notorious for not playing well with AutoDiff systems, unless JAX tackled this problem?

**Comment by btaba** [comment]
JAX is [auto-differentiable](https://docs.jax.dev/en/latest/automatic-differentiation.html), but not much effort beyond that was made to make gradients play nicely, YMMV
